src/iceberg_destination.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use arrow::array::RecordBatch;
+use arrow_schema::{Field, Schema, SchemaRef};
+use futures::{pin_mut, StreamExt, TryStream};
+use iceberg::io::FileIO;
+use iceberg::spec::{
+    DataContentType, DataFileFormat, FormatVersion, Manifest, ManifestContentType, ManifestEntry,
+    ManifestFile, ManifestListWriter, ManifestMetadata, ManifestStatus, ManifestWriter, Operation,
+    PartitionSpec, Snapshot, Struct, Summary, TableMetadata, TableMetadataBuilder,
+};
+use iceberg::writer::file_writer::location_generator::{
+    DefaultFileNameGenerator, DefaultLocationGenerator,
+};
+use iceberg::writer::file_writer::{FileWriter, FileWriterBuilder, ParquetWriterBuilder};
+use iceberg::TableCreation;
+use log::info;
+use parquet::arrow::PARQUET_FIELD_ID_META_KEY;
+use parquet::file::properties::WriterProperties;
+use url::Url;
+use uuid::Uuid;
+
+use crate::error::DataLoadingError;
+
+pub async fn record_batches_to_iceberg(
+    record_batch_stream: impl TryStream<Item = Result<RecordBatch, DataLoadingError>>,
+    arrow_schema: SchemaRef,
+    target_url: Url,
+) -> Result<(), DataLoadingError> {
+    pin_mut!(record_batch_stream);
+
+    let mut file_io_props: Vec<(String, String)> = vec![];
+    if let Ok(aws_endpoint) = std::env::var("AWS_ENDPOINT") {
+        file_io_props.push(("s3.endpoint".to_string(), aws_endpoint));
+    }
+
+    let file_io = FileIO::from_path(target_url.clone())?
+        .with_props(file_io_props)
+        .build()?;
+
+    // Assign sequential field IDs starting from 1 to the fields
+    let mut field_id_counter = 1;
+    let fields: Vec<Field> = arrow_schema
+        .fields
+        .iter()
+        .map(|field_ref| {
+            let mut field: Field = (**field_ref).clone();
+            let mut metadata = field_ref.metadata().clone();
+            metadata.insert(
+                PARQUET_FIELD_ID_META_KEY.to_owned(),
+                field_id_counter.to_string(),
+            );
+            field_id_counter += 1;
+            field.set_metadata(metadata);
+            field
+        })
+        .collect();
+
+    let cloned_arrow_schema = Schema::new_with_metadata(fields, arrow_schema.metadata.clone());
+
+    let iceberg_schema = iceberg::arrow::arrow_schema_to_schema(&cloned_arrow_schema)?;
+
+    let table_creation = TableCreation::builder()
+        .name("dummy_name".to_string()) // Required by TableCreationBuilder. Doesn't affect output
```

Comment:
Presumably this name is stored in table metadata? If so how about using something that conveys at least some information (e.g. signifying that the table was created by lakehouse-loader, timestamp/source) or adding a new name param to the CLI?

Reply:
No, it doesn't get persisted in table metadata or anywhere else